### ISSUE TITLE
sqlean flags `lower()` as non-deterministic

### ISSUE THREAD

**Issue by mikeschinkel** [nalgeon/sqlean#109]
Running `sqlean` version `3.45.0` on macOS I get this error:

```
Parse error near line 120: non-deterministic functions prohibited in generated columns
      domain       VARCHAR(24) GENERATED ALWAYS AS (lower(substr(email, instr(em
                                      error here ---^
```

The DDL for the table is:

```sql
create table email
(
   email_id     INTEGER
      primary key autoincrement,
   source       CHAR(1), -- A=Account,C=Contact
   source_email VARCHAR(48),
   is_valid     CHAR(1),
   email        VARCHAR(48),
   domain       VARCHAR(24) GENERATED ALWAYS AS (lower(substr(email, instr(email, '@') + 1)))
);
```
I am assuming that `lower()`, `substr()`, and `instr()` are all deterministic, right?  

This DDL works in `sqlite3` `v3.37.0`.

**Comment by nalgeon** [comment]
Sure! Fixed in [0.21.9](https://github.com/nalgeon/sqlean/releases/0.21.9).

**Comment by mikeschinkel** [comment]
Wow, talk about service!  Thanks.

P.S. I have another one for you, slightly harder I expect.  Issue pending.
